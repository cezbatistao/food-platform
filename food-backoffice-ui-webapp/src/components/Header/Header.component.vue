<template>
  <div class="container">
    <header class="d-flex flex-wrap justify-content-center py-3 mb-4 border-bottom">
      <a href="/" class="d-flex align-items-center mb-3 mb-md-0 me-md-auto text-dark text-decoration-none">
        <i style="padding-right: 15px; padding-left: 15px" className="fas fa-desktop fa-2x"></i>
        <span class="fs-4">Food Plataform Backoffice</span>
      </a>
      <ul class="nav nav-pills">
        <li class="nav-item">
          <router-link class="nav-link" active-class="active" to="/">Home</router-link>
        </li>
        <li class="nav-item">
          <router-link class="nav-link" active-class="active" to="/restaurant">Restaurante</router-link>
        </li>
      </ul>
    </header>
  </div>
</template>

<script lang="ts">
  import { defineComponent } from 'vue'

  export default defineComponent({
    name: 'header'
  })
</script>