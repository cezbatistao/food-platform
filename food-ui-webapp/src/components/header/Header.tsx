const Header = () => {
  return (
    <div className="container">
    <header className="d-flex flex-wrap justify-content-center py-3 mb-4 border-bottom">
      <a href="/" className="d-flex align-items-center mb-3 mb-md-0 me-md-auto text-dark text-decoration-none">
        <i style={{ paddingRight: "15px" }} className="fas fa-utensils fa-2x"></i>
        <span className="fs-4">Food Platform</span>
      </a>
      <ul className="nav nav-pills">
        <li className="nav-item">
          <i className="fas fa-shopping-basket fa-2x"></i>
        </li>
      </ul>
    </header>
  </div>
  );
}

export default Header;