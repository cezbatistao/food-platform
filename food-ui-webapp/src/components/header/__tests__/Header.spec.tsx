import { shallow } from "enzyme";

import Header from "../Header";

describe('Header', () => {

  it("renders the title app name", () => {
      const result = shallow(<Header />);
      expect(result.text()).toEqual('Food Platform');
  });

  it("renders the heading", () => {
    const result = shallow(<Header />);
    expect(result).toMatchSnapshot();
  });

});