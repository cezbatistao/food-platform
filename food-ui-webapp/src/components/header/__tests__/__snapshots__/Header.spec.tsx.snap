// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders the heading 1`] = `
<div
  className="container"
>
  <header
    className="d-flex flex-wrap justify-content-center py-3 mb-4 border-bottom"
  >
    <a
      className="d-flex align-items-center mb-3 mb-md-0 me-md-auto text-dark text-decoration-none"
      href="/"
    >
      <i
        className="fas fa-utensils fa-2x"
        style={
          Object {
            "paddingRight": "15px",
          }
        }
      />
      <span
        className="fs-4"
      >
        Food Platform
      </span>
    </a>
    <ul
      className="nav nav-pills"
    >
      <li
        className="nav-item"
      >
        <i
          className="fas fa-shopping-basket fa-2x"
        />
      </li>
    </ul>
  </header>
</div>
`;
